MediaBrowser.Providers/Chapters/ChapterManager.cs

```diff
@@ -2,25 +2,52 @@
 
 using System;
 using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Jellyfin.Data.Entities;
+using Jellyfin.Server.Implementations;
 using MediaBrowser.Controller.Chapters;
-using MediaBrowser.Controller.Persistence;
-using MediaBrowser.Model.Entities;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
 
 namespace MediaBrowser.Providers.Chapters
 {
     public class ChapterManager : IChapterManager
     {
-        private readonly IItemRepository _itemRepo;
+        private readonly IDbContextFactory<LibraryDbContext> _provider;
+        private readonly ILogger<ChapterManager> _logger;
 
-        public ChapterManager(IItemRepository itemRepo)
+        public ChapterManager(IDbContextFactory<LibraryDbContext> provider, ILogger<ChapterManager> logger)
         {
-            _itemRepo = itemRepo;
+            _provider = provider;
+            this._logger = logger;
         }
 
         /// <inheritdoc />
-        public void SaveChapters(Guid itemId, IReadOnlyList<ChapterInfo> chapters)
+        public async void SaveChapters(Guid itemId, IReadOnlyList<ChapterInfo> chapters)
```

Comment:
Avoid `async void`

Reply:
Can I ask why?

Reply:
It swallows exceptions and is considered a dirty hack thats only existance is allowed on eventhandlers where there is no other option.

Reply:
The exception part makes sense. I’ll fix it in the evening. 🙂

Reply:
@JPVenson Would consider `async Task` better?

Reply:
Yes that also means you have to await the task at the caller